File: src/2894.divisible-and-non-divisible-sums-difference.3172/solution.ts
function differenceOfSums (n: number, m: number): number {
    let result = 0;
    for (let i = 1; i <= n; i++) {
        if (i % m === 0) {
            result -= i;
        } else {
            result += i;
        }
    }
    return result;
}

File: src/2894.divisible-and-non-divisible-sums-difference.3172/solution.java
class Solution {
    public int differenceOfSums(int n, int m) {
        int result = 0;
        while (n > 0) {
            if (n % m == 0) {
                result -= n;
            } else {
                result += n;
            }
            n--;
        }
        return result;
    }
}